### PR TITLE
transaction model

### PR DESCRIPTION
**Pull request by Gullu143**
create a new transaction model

### CODE REVIEW

File: app/assets/javascripts/transactions.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (saqib1324) [inline app/assets/javascripts/transactions.coffee:1]: delete this file, we don't need this

File: app/assets/stylesheets/transactions.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the transactions controller here.
```
Comment (saqib1324) [inline app/assets/stylesheets/transactions.scss:1]: delete this file, we don't need this

File: app/controllers/transactions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def new
+  	@transaction = Transaction.new
+  end
+
+  def create
+  	@transaction = Transaction.new(transation_params)
+  	if @transaction.save
+  		redirect_to show_representative_path, notice: 'transaction successfully created'
```
Comment (saqib1324) [inline app/controllers/transactions_controller.rb:13]: `show_representative_path` requires an `:id` parameter that tells the router that which representatives' show page we want to open. and here we want to open the show page of `@transaction.representative`, so replace path with:
`show_representative_path(@transaction.representative)`

File: app/helpers/transactions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module TransactionsHelper
```
Comment (saqib1324) [inline app/helpers/transactions_helper.rb:1]: remove this file..

File: app/models/representative.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 class Representative < ApplicationRecord
   belongs_to :user
+  has_many :transactions
```
Comment (saqib1324) [inline app/models/representative.rb:3]: add dependent destroy here.

File: app/views/transactions/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<h1>Transactions#edit</h1>
```
Comment (saqib1324) [inline app/views/transactions/edit.html.erb:1]: remove this file

File: app/views/transactions/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
+<p>Find me in app/views/transactions/edit.html.erb</p>
```
Comment (saqib1324) [inline app/views/transactions/edit.html.erb:2]: remove this file

File: app/views/transactions/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+<h1>Transactions#new</h1>
```
Comment (saqib1324) [inline app/views/transactions/new.html.erb:1]: remove this file

File: config/routes.rb [hunk 9]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   resources :representatives
+  resources :transactions
```
Comment (saqib1324) [inline config/routes.rb:3]: we only want to create transactions, so we don't need all routes, we just need routes of create. replace this line with:
`resources :transactions, only: :create`

File: test/controllers/transactions_controller_test.rb [hunk 10]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
```
Comment (saqib1324) [inline test/controllers/transactions_controller_test.rb:2]: remove this file

File: test/fixtures/transactions.yml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+one:
+  status: 1
+  amount: 1.5
```
Comment (saqib1324) [inline test/fixtures/transactions.yml:5]: remove this file

File: test/models/transaction_test.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class TransactionTest < ActiveSupport::TestCase
```
Comment (saqib1324) [inline test/models/transaction_test.rb:3]: remove this file

File: app/controllers/transactions_controller.rb [hunk 13]
```diff
@@ -0,0 +1,23 @@
+class TransactionsController < ApplicationController
+
+  def edit
+  end
+
+  def new
```
Comment (saqib1324) [inline app/controllers/transactions_controller.rb:6]: we only need create action Transactions, remove all other actions.

File: app/assets/javascripts/transactions.coffee (hunk 1 above)
Comment (Gullu143) [reply app/assets/javascripts/transactions.coffee:1]: file deleted

File: app/assets/stylesheets/transactions.scss (hunk 2 above)
Comment (Gullu143) [reply app/assets/stylesheets/transactions.scss:1]: file deleted

File: app/controllers/transactions_controller.rb (hunk 3 above)
Comment (Gullu143) [reply app/controllers/transactions_controller.rb:13]: replaced code

File: app/helpers/transactions_helper.rb (hunk 4 above)
Comment (Gullu143) [reply app/helpers/transactions_helper.rb:1]: file deleted

File: app/models/representative.rb (hunk 5 above)
Comment (Gullu143) [reply app/models/representative.rb:3]: added dependent destroy

File: app/views/transactions/edit.html.erb (hunk 6 above)
Comment (Gullu143) [reply app/views/transactions/edit.html.erb:1]: file removed

File: app/views/transactions/edit.html.erb (hunk 7 above)
Comment (Gullu143) [reply app/views/transactions/edit.html.erb:2]: file removed

File: app/views/transactions/new.html.erb (hunk 8 above)
Comment (Gullu143) [reply app/views/transactions/new.html.erb:1]: file removed

File: config/routes.rb (hunk 9 above)
Comment (Gullu143) [reply config/routes.rb:3]: replaced

File: test/controllers/transactions_controller_test.rb (hunk 10 above)
Comment (Gullu143) [reply test/controllers/transactions_controller_test.rb:2]: file removed

File: test/fixtures/transactions.yml (hunk 11 above)
Comment (Gullu143) [reply test/fixtures/transactions.yml:5]: file removed

File: test/models/transaction_test.rb (hunk 12 above)
Comment (Gullu143) [reply test/models/transaction_test.rb:3]: file removed

File: app/controllers/transactions_controller.rb (hunk 13 above)
Comment (Gullu143) [reply app/controllers/transactions_controller.rb:6]: other actions removed

File: config/database.yml [hunk 14]
```diff
@@ -24,6 +24,8 @@ default: &default
 development:
   <<: *default
   database: represent_development
+  username: rails_dev
```
Comment (saqib1324) [inline config/database.yml:27]: add `/config/database.yml` to your .gitignore file so that your database.yml is not pushed to git.

Comment (Gullu143) [reply config/database.yml]: database name deleted

**Pull request merged**